### PR TITLE
Add bip47 message classes and verification tests

### PR DESCRIPTION
**Pull request by kmels**

### CODE REVIEW

File: include/opentxs-proto/verify/Bip47Address.hpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#ifndef OPENTXS_PROTO_BIP47_ADDRESS_HPP
+#define OPENTXS_PROTO_BIP47_ADDRESS_HPP
+
+#include "VerifyBlockchain.hpp"
+
+namespace opentxs
+{
+namespace proto
+{
+bool CheckProto_1(const Bip47Address& address, const bool silent);
```
Comment (justusranvier) [inline include/opentxs-proto/verify/Bip47Address.hpp]: For consistency with the validation macros, the first argument should be named "input" for all of these functions

File: include/opentxs-proto/verify/VerifyBlockchain.hpp [hunk 2]
```diff
@@ -59,6 +64,18 @@ static const VersionMap Bip44AccountAllowedBip44Address = {
 static const VersionMap Bip44AccountAllowedHDPath = {
     {1, {1, 1}},
 };
+static const VersionMap Bip47AddressAllowed = {
```
Comment (justusranvier) [inline include/opentxs-proto/verify/VerifyBlockchain.hpp]: The format for these variables is "FooAllowedBar" where "Foo" is the parent class and "Bar" is the embedded child class.

File: src/verify/bip47address/Bip47Address_1.cpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+#define PROTO_NAME "bip47 address"
+
+namespace opentxs
+{
+namespace proto
+{
+
+bool CheckProto_1(const Bip47Address& input, const bool silent)
+{
+    if (false == input.has_address()) {
```
Comment (justusranvier) [inline src/verify/bip47address/Bip47Address_1.cpp]: Use CHECK_IDENTIFIER macro here

File: src/verify/bip47address/Bip47Address_1.cpp [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        if (MIN_PLAUSIBLE_SCRIPT > input.address().size()) {
+            FAIL("bip47 address", "invalid address")
+        }
+
+        if (MAX_PLAUSIBLE_SCRIPT < input.address().size()) {
+            FAIL("bip47 address", "invalid address")
+        }
+    }
+
+    for (const auto& txid : input.incoming()) {
```
Comment (justusranvier) [inline src/verify/bip47address/Bip47Address_1.cpp]: use CHECK_IDENTIFIERS macro here

File: src/verify/bip47address/Bip47Address_1.cpp [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            FAIL("bip47 address", "invalid txid")
+        }
+    }
+
+    return true;
+}
+
+bool CheckProto_2(const Bip47Address& input, const bool silent)
+{
+    UNDEFINED_VERSION("bip47 address", 2)
```
Comment (justusranvier) [inline src/verify/bip47address/Bip47Address_1.cpp]: use UNDEFINED_VERSION2 macro here

File: src/verify/bip47chain/Bip47Chain_1.cpp [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    if (false == validChain) { FAIL("bip47 chain", "invalid type") }
+
+    CHECK_SUBOBJECTS(channel, Bip47ChannelAllowed);
+      
+    return true;
+}
+
+bool CheckProto_2(const Bip47Chain& input, const bool silent)
+{
+    UNDEFINED_VERSION("bip47 chain", 2)
```
Comment (justusranvier) [inline src/verify/bip47chain/Bip47Chain_1.cpp]: UNDEFINED_VERSION2

File: include/opentxs-proto/verify/VerifyBlockchain.hpp (hunk 2 above)
Comment (kmels) [reply include/opentxs-proto/verify/VerifyBlockchain.hpp]: Changed

File: src/verify/bip47address/Bip47Address_1.cpp (hunk 3 above)
Comment (kmels) [reply src/verify/bip47address/Bip47Address_1.cpp]: Yes.. changed!

File: src/verify/bip47chain/Bip47Chain_1.cpp (hunk 6 above)
Comment (kmels) [reply src/verify/bip47chain/Bip47Chain_1.cpp]: Updated together with other classes too.

**Pull request merged**